lib/internal/util.js

```diff
@@ -1,18 +1,65 @@
 'use strict';
 
-exports.printDeprecationMessage = function(msg, warned) {
+const util = require('util');
+const warnedLocations = new Set();
+
+function getActualCallerLocation() {
+  // If external, we split five times, because stack trace will be like this
+  //   1. Error:
+  //   2. getActualCallerLocation
+  //   3. printDeprecationMessage
+  //   4. `deprecated` function in `util.deprecate`
+  //   5. The file which actually called `util.deprecate`
+  // So, we pick the fifth entry in the stack trace and parse it
+  const location = (new Error()).stack.split('\n', 5).pop();
```

Comment:
Can you? I wouldn't expect it to work because util.deprecate just returns the function that is effectively in the stack. You would have to pass this function but it's not possible.